### PR TITLE
Aristos espionage add on

### PR DESCRIPTION
**Pull request by Azurency**
Update from Astog, you can read more about it here https://github.com/Azurency/CQUI_Community-Edition/issues/234

### CODE REVIEW

**Comment by ricanuck** [comment]
Credit where it is due. Right? It is not an update FROM astog. Nor FROM you.

**Review by ricanuck — CHANGES_REQUESTED** [review]
Credit has to go where it is due, no matter how much any person may disagree with any other. THAT is how open source works. "Stealing" code without shame and due credit is not acceptable (even if it is a few lines).

File: UI/EspionageSupport.lua [hunk 1]
```diff
@@ -129,7 +129,7 @@ function hasDistrict(city:table, districtType:string)
     local hasDistrict:boolean = false;
     local cityDistricts:table = city:GetDistricts();
     for i, district in cityDistricts:Members() do
-        if district:IsComplete() then
+        if district:IsComplete() and not district:IsPillaged() then
```
Comment (ricanuck) [inline UI/EspionageSupport.lua]: Insert for due credit:
--ARISTOS: to only show available and valid targets in each city, both for espionage overview and selector

File: UI/PartialScreens/EspionageOverview.lua [hunk 2]
```diff
@@ -421,20 +421,21 @@ function AddDistrictIcon(stackControl:table, city:table, districtType:string)
...
+    --         if districtInfo.DistrictType == districtType then
+    --             hasDistrict = true;
+    --             toolTipString = Locale.Lookup(districtInfo.Name);
+    --         end
+    --     end
+    -- end
 
     -- We're manipulating the alpha to hide each element so they maintain their stack positions
-    if hasDistrict then
+    if hasDistrict(city, districtType) then
```
Comment (ricanuck) [inline UI/PartialScreens/EspionageOverview.lua]: Insert for due credit:
--ARISTOS: make use of the espionagesupport.lua funtion, more efficient and has been fixed to only show valid targets

File: UI/PartialScreens/EspionageOverview.lua [hunk 3]
```diff
@@ -421,20 +421,21 @@ function AddDistrictIcon(stackControl:table, city:table, districtType:string)
     local toolTipString:string = "";
 
     -- Check if we have this district
-    local hasDistrict:boolean = false;
```
Comment (ricanuck) [inline UI/PartialScreens/EspionageOverview.lua]: Erase all this, no need to leave it block commented.

**Comment by Azurency** [comment]
Wow, I'm really sorry, I wrote Astog without even noticing (might be because I was in his repository). Of course I meant you Aristos, I corrected in the code also with your credits.

**Comment by ricanuck** [comment]
Thanks.

**Comment by Azurency** [comment]
We've merged this changes in CQUI.

**Pull request merged**